components/script/dom/fontface.rs

```diff
@@ -0,0 +1,568 @@
+        implement_setter_for_descriptor!(self, variationSettings, value)
+    }
+
+    /// <https://drafts.csswg.org/css-font-loading/#dom-fontface-display>
+    fn Display(&self) -> DOMString {
+        self.descriptors.borrow().display.clone()
+    }
+
+    /// <https://drafts.csswg.org/css-font-loading/#dom-fontface-display>
+    fn SetDisplay(&self, value: DOMString) -> ErrorResult {
+        implement_setter_for_descriptor!(self, display, value)
+    }
+
+    /// <https://drafts.csswg.org/css-font-loading/#dom-fontface-ascentoverride>
+    fn AscentOverride(&self) -> DOMString {
+        self.descriptors.borrow().ascentOverride.clone()
+    }
+
+    /// <https://drafts.csswg.org/css-font-loading/#dom-fontface-ascentoverride>
+    fn SetAscentOverride(&self, value: DOMString) -> ErrorResult {
+        implement_setter_for_descriptor!(self, ascentOverride, value)
+    }
+
+    /// <https://drafts.csswg.org/css-font-loading/#dom-fontface-descentoverride>
+    fn DescentOverride(&self) -> DOMString {
+        self.descriptors.borrow().descentOverride.clone()
+    }
+
+    /// <https://drafts.csswg.org/css-font-loading/#dom-fontface-descentoverride>
+    fn SetDescentOverride(&self, value: DOMString) -> ErrorResult {
+        implement_setter_for_descriptor!(self, descentOverride, value)
+    }
+
+    /// <https://drafts.csswg.org/css-font-loading/#dom-fontface-linegapoverride>
+    fn LineGapOverride(&self) -> DOMString {
+        self.descriptors.borrow().lineGapOverride.clone()
+    }
+
+    /// <https://drafts.csswg.org/css-font-loading/#dom-fontface-linegapoverride>
+    fn SetLineGapOverride(&self, value: DOMString) -> ErrorResult {
+        implement_setter_for_descriptor!(self, lineGapOverride, value)
+    }
+
+    /// <https://drafts.csswg.org/css-font-loading/#dom-fontface-status>
+    fn Status(&self) -> FontFaceLoadStatus {
+        self.status.get()
+    }
+
+    /// The load() method of FontFace forces a url-based font face to request its font data and
+    /// load. For fonts constructed from a buffer source, or fonts that are already loading or
+    /// loaded, it does nothing.
+    /// <https://drafts.csswg.org/css-font-loading/#font-face-load>
+    fn Load(&self) -> Rc<Promise> {
+        let Some(sources) = self.urls.borrow_mut().take() else {
+            // Step 2. If font face’s [[Urls]] slot is null, or its status attribute is anything
+            // other than "unloaded", return font face’s [[FontStatusPromise]] and abort these
+            // steps.
+            return self.font_status_promise.clone();
+        };
+
+        // FontFace must not be loaded at this point as `self.urls` is not None, implying `Load`
+        // wasn't called already. In our implementation, `urls` is set after parsing, so it
+        // cannot be `Some` if the status is `Error`.
+        debug_assert_eq!(self.status.get(), FontFaceLoadStatus::Unloaded);
+
+        let trusted = Trusted::new(self);
+        let task_source = self
```

Comment:
If you move `let global = self.global()` above this line you can avoid calling `self.global()` here.